foursight_core/routes.py

```diff
@@ -0,0 +1,231 @@
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}', methods=['GET'])
+    def get_check_route(environ, check):
+        """
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):
+            return app.core.run_get_check(environ, check, None)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}', methods=['PUT'])
+    def put_check_route(environ, check):
+        """
+        Take a PUT request. Body of the request should be a json object with keys
+        corresponding to the fields in CheckResult, namely:
+        title, status, description, brief_output, full_output, uuid.
+        If uuid is provided and a previous check is found, the default
+        behavior is to append brief_output and full_output.
+
+        Protected route
+        """
+        request = app.current_request
+        if app.core.check_authorization(request.to_dict(), environ):
+            put_data = request.json_body
+            return app.core.run_put_check(environ, check, put_data)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'environments/{environ}', methods=['PUT'])
+    def put_environment(environ):
+        """
+        Take a PUT request that has a json payload with 'fourfront' (ff server)
+        and 'es' (es server).
+        Attempts to generate an new environment and runs all checks initially
+        if successful.
+
+        Protected route
+        """
+        request = app.current_request
+        if app.core.check_authorization(request.to_dict(), environ):
+            env_data = request.json_body
+            return app.core.run_put_environment(environ, env_data)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'environments/{environ}', methods=['GET'])
+    def get_environment_route(environ):
+        """
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):
+            return app.core.run_get_environment(environ)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'environments/{environ}/delete', methods=['DELETE'])
+    def delete_environment(environ):
+        """
+        Takes a DELETE request and purges the foursight environment specified by 'environ'.
+        NOTE: This only de-schedules all checks, it does NOT wipe data associated with this
+        environment - that can only be done directly from S3 (for safety reasons).
+
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):  # TODO (C4-138) Centralize authorization check
+            return app.core.run_delete_environment(environ)
+        else:
+            return app.core.forbidden_response()
```

Comment:
Same deal here - should be turned off, don't want users messing with this most likely at least right now.

Reply:
All set (commented out).